test/test_tensorclass_nofuture.py

```diff
@@ -354,6 +354,47 @@ class MyDataNested:
     ):
         data[:2] = data3[:2]
 
+    # regression test PR #203
+    # We should be able to set tensors items with MemmapTensors and viceversa
+    @tensorclass
+    class MyDataMemMap1:
+        x: torch.Tensor
+        y: MemmapTensor
+
+    @tensorclass
+    class MyDataMemMap2:
+        x: MemmapTensor
+        y: torch.Tensor
+
+    data1 = MyDataMemMap1(
+        x=torch.zeros(3, 4, 5),
+        y=MemmapTensor.from_tensor(torch.zeros(3, 4, 5)),
+        batch_size=[3, 4],
+    )
+    data2_wrong = MyDataMemMap2(
+        x=MemmapTensor.from_tensor(torch.ones(3, 4, 5)),
+        y=torch.ones(3, 4, 5),
+        batch_size=[3, 4],
+    )
+    with pytest.raises(ValueError, match="__setitem__ is only allowed for same-class assignement"):
+        data1[:2] = data2_wrong[:2]
+    with pytest.raises(ValueError, match="__setitem__ is only allowed for same-class assignement"):
```

Comment:
The fact that this raises and the following case not is a bit strange. @vmoens wdyt?

Reply:
I just tried and on my side both tests pass

Reply:
they pass *without* the pytest.raise?

Reply:
I'm confused, the first should work with `pytest.raises` but not the second, which should not raise any exception. Is it what you're saying? Why would it be the case?

Reply:
no, i am comparing this two "raises" checks with the case at lines 389 and following. The only difference between data2_wrong and data2 is the class name. I think that is similar to what you mention about inheritance in the other comment thread.